src/app/shared/components/dark-mode-toggle/dark-mode-toggle.component.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { DOCUMENT } from '@angular/common';
+import { MediaMatcher } from '@angular/cdk/layout';
+import { StorageService } from '../../services/storage.service';
+
+@Component({
+  selector: 'app-dark-mode-toggle',
+  templateUrl: './dark-mode-toggle.component.html',
+  styleUrls: ['./dark-mode-toggle.component.scss'],
+})
+export class DarkModeToggleComponent implements OnInit {
+  isDarkMode: boolean;
+
+  constructor(
+    @Inject(DOCUMENT)
+    private readonly document: Document,
+    private readonly mediaMatcher: MediaMatcher,
+    private readonly storageService: StorageService,
+  ) {}
+
+  ngOnInit() {
+    const userPrefersDark =
+      this.mediaMatcher.matchMedia &&
+      this.mediaMatcher.matchMedia('(prefers-color-scheme: dark)').matches;
+    // In case the user has used the toggle button, we prioritize it over the
+    // system settings
+    this.setDarkMode(this.getUserSettingsIsDarkMode() || userPrefersDark);
```

Comment:
This logic doesn't seem to work as expected (explicit setting is not memorized)

Reply:
On this part of the logic to me it seems to work correctly, except renaming the names, I left everything as Livio had configured 3 years ago. 😅 Let me know what you mean about this part. 🙂